packages/client/src/build-id-types.ts

```diff
@@ -0,0 +1,137 @@
+import { temporal } from '@temporalio/proto';
+
+/**
+ * Operations that can be passed to {@link WorkflowClient.updateWorkerBuildIdCompatability}.
+ *
+ * @experimental
+ */
+export type BuildIdOperation =
+  | NewIdInNewDefaultSet
+  | NewCompatibleVersion
+  | PromoteSetByBuildId
+  | PromoteBuildIdWithinSet
+  | MergeSets;
+
+/**
+ * This operation adds a new Build Id into a new set, which will be used as the default set for
+ * the queue. This means all new workflows will start on this Build Id.
+ *
+ * @experimental
+ */
+interface NewIdInNewDefaultSet {
+  operation: 'NEW_ID_IN_NEW_DEFAULT_SET';
+  buildId: string;
+}
+
+/**
+ * This operation adds a new Build Id into an existing compatible set. The newly added ID becomes
+ * the default for that compatible set, and thus new workflow tasks for workflows which have been
+ * executing on workers in that set will now start on this new Build ID.
+ *
+ * @experimental
+ */
+interface NewCompatibleVersion {
+  operation: 'NEW_COMPATIBLE_VERSION';
+  // buildId The Build Id to add to an existing compatible set.
+  buildId: string;
+  // A Build Id which must already be defined on the task queue, and is used to
+  // find the compatible set to add the new ID to.
+  existingCompatibleBuildId: string;
+  // If set to true, the targeted set will also be promoted to become the
+  // overall default set for the queue.
+  makeSetDefault?: boolean;
+}
+
+/**
+ * This operation promotes a set of compatible Build IDs to become the current
```

Comment:
```suggestion
 * This operation promotes a set of compatible Build Ids to become the current
```

Reply:
We haven't been consistent in the reset of the codebase, looks like we use `Id` in schedules docs. And a mix of them in workflow options :(